leaves/commits.go

```diff
@@ -0,0 +1,229 @@
+// to this list. Also used by core.Registry to build the global map of providers.
+func (ca *CommitsAnalysis) Provides() []string {
+	return []string{}
+}
+
+// Requires returns the list of names of entities which are needed by this PipelineItem.
+// Each requested entity will be inserted into `deps` of Consume(). In turn, those
+// entities are Provides() upstream.
+func (ca *CommitsAnalysis) Requires() []string {
+	arr := [...]string{
+		identity.DependencyAuthor, items.DependencyLanguages, items.DependencyLineStats}
+	return arr[:]
+}
+
+// ListConfigurationOptions returns the list of changeable public properties of this PipelineItem.
+func (ca *CommitsAnalysis) ListConfigurationOptions() []core.ConfigurationOption {
+	return nil
+}
+
+// Configure sets the properties previously published by ListConfigurationOptions().
+func (ca *CommitsAnalysis) Configure(facts map[string]interface{}) error {
+	if val, exists := facts[identity.FactIdentityDetectorReversedPeopleDict].([]string); exists {
+		ca.reversedPeopleDict = val
+	}
+	return nil
+}
+
+// Flag for the command line switch which enables this analysis.
+func (ca *CommitsAnalysis) Flag() string {
+	return "commits-stat"
+}
+
+// Description returns the text which explains what the analysis is doing.
+func (ca *CommitsAnalysis) Description() string {
+	return "Extracts statistics for each commit."
+}
+
+// Initialize resets the temporary caches and prepares this PipelineItem for a series of Consume()
+// calls. The repository which is going to be analysed is supplied as an argument.
+func (ca *CommitsAnalysis) Initialize(repository *git.Repository) error {
+	ca.OneShotMergeProcessor.Initialize()
+	return nil
+}
+
+// Consume runs this PipelineItem on the next commit data.
+// `deps` contain all the results from upstream PipelineItem-s as requested by Requires().
+// Additionally, DependencyCommit is always present there and represents the analysed *object.Commit.
+// This function returns the mapping with analysis results. The keys must be the same as
+// in Provides(). If there was an error, nil is returned.
+func (ca *CommitsAnalysis) Consume(deps map[string]interface{}) (map[string]interface{}, error) {
+	if !ca.ShouldConsumeCommit(deps) {
+		return nil, nil
+	}
+	if deps[core.DependencyIsMerge].(bool) {
+		return nil, nil
+	}
+
+	commit := deps[core.DependencyCommit].(*object.Commit)
+	author := deps[identity.DependencyAuthor].(int)
+	lineStats := deps[items.DependencyLineStats].(map[object.ChangeEntry]items.LineStats)
+	langs := deps[items.DependencyLanguages].(map[plumbing.Hash]string)
+
+	cs := CommitStat{
+		Hash:   commit.Hash.String(),
+		When:   commit.Author.When.Unix(),
+		Author: author,
+	}
+
+	for entry, stats := range lineStats {
+		cs.Files = append(cs.Files, FileStat{
+			Name:      entry.Name,
+			Language:  langs[entry.TreeEntry.Hash],
+			LineStats: stats,
+		})
+	}
+
+	ca.commits = append(ca.commits, &cs)
+
```

Comment:
Not sure if having so many blank lines is a feature or a bug. I personally don't like having empty lines, but I am fine if this is your own coding style.

Reply:
Yes. It's the style I use but thank you for pointing on it! I removed blank lines to follow the project's code style.